modules/videoio/src/cap_ffmpeg_impl.hpp

```diff
@@ -967,6 +967,25 @@ bool CvCapture_FFMPEG::open( const char* _filename )
         enc->thread_count = get_number_of_cpus();
 //#endif
 
+        AVDictionaryEntry* avdiscard_entry = av_dict_get(dict, "avdiscard", NULL, 0);
+
+        if (avdiscard_entry != 0) {
+            if(strcmp(avdiscard_entry->value, "all") == 0)
+                enc->skip_frame = AVDISCARD_ALL;
+            else if (strcmp(avdiscard_entry->value, "bidir") == 0)
+                enc->skip_frame = AVDISCARD_BIDIR;
+            else if (strcmp(avdiscard_entry->value, "default") == 0)
+                enc->skip_frame = AVDISCARD_DEFAULT;
+            else if (strcmp(avdiscard_entry->value, "none") == 0)
+                enc->skip_frame = AVDISCARD_NONE;
+            else if (strcmp(avdiscard_entry->value, "nonintra") == 0)
+                enc->skip_frame = AVDISCARD_NONINTRA;
+            else if (strcmp(avdiscard_entry->value, "nonkey") == 0)
+                enc->skip_frame = AVDISCARD_NONKEY;
+            else if (strcmp(avdiscard_entry->value, "nonref") == 0)
+                enc->skip_frame = AVDISCARD_NONREF;
```

Comment:
What FFmpeg / libav versions support that? (at least compilation)

Need to properly guard code through "#if"s.

BTW, access to `OPENCV_FFMPEG_CAPTURE_OPTIONS` is guarded by
```
#if LIBAVFORMAT_BUILD >= CALC_FFMPEG_VERSION(52, 111, 0)
```

Reply:
@alalek All options (except `AVDISCARD_NONINTRA`) are in use since ffmpeg 0.7 (libavcodec 52.123.0)

First use of `AVDISCARD_NONINTRA` is in version 2.3 (libavcodec 55.69.100).

I added some checks on LIBAVCODEC_BUILD version in my last commit, and squashed previous ones.